src/argilla/client/feedback/dataset/remote/filtered.py

```diff
@@ -52,14 +52,15 @@ def __init__(
             [metadata_filter.query_string for metadata_filter in metadata_filters] if metadata_filters else None
         )
 
-    def __len__(self) -> None:
+    def __len__(self) -> int:
         warnings.warn(
             "The `records` of a filtered dataset in Argilla are being lazily loaded"
             " and len computation may add undesirable extra computation. You can fetch"
             "records using\n`ds.pull()`\nor iterate over results to know the length of the result:\n"
             "`records = [r for r in ds.records]\n",
             stacklevel=1,
         )
+        return 0
```

Comment:
I think this can be misleading, I'm more comfortable with not retuning anything i.e. `None`